### PR TITLE
Add Retry Configs to EgressConfig

### PR DESCRIPTION
**Pull request by pierDipi**
This PR adds retry configs to the contract configuration in preparation
of supporting the delivery spec: #116

Signed-off-by: Pierangelo Di Pilato <pierangelodipilato@gmail.com>

Part of #116

## Proposed Changes

- Add Retry Configs to EgressConfig

**Release Note**

```release-note
- 🧽 Update or clean up current behavior
The data plane contract contains retry configurations.
```

### CODE REVIEW

**Comment by pierDipi** [comment]
The linter checks trailing whitespaces in the generated code.

File: proto/def/contract.proto [hunk 1]
```diff
@@ -16,9 +16,35 @@ message Filter {
...
+  Linear = 1;
+}
+
 message EgressConfig {
   // Dead letter is where the event is sent when something goes wrong
   string deadLetter = 1;
+
+  // retry is the minimum number of retries the sender should attempt when
+  // sending an event before moving it to the dead letter sink.
+  uint32 retry = 2;
```
Comment (slinkydeveloper) [inline proto/def/contract.proto:35]: no retries is identified as `retry = 0` right?

**Review by slinkydeveloper — COMMENTED** [review]
Just a question

File: proto/def/contract.proto (hunk 1 above)
Comment (pierDipi) [reply proto/def/contract.proto:37]: Done: https://github.com/knative-sandbox/eventing-kafka-broker/pull/258/commits/7bba73d648a70af57f9a88ab259ff472415b9ef0

**Comment by slinkydeveloper** [comment]
Unit test failure is a general flakyness, not related to this one right?

**Comment by pierDipi** [comment]
Yes, micrometer metrics use atomic fields, and they never fail in Prow AFAIK.

**Pull request merged**